news-bundle/src/Feed/FeedProvider.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\Feed;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\Model\Collection;
+use Contao\NewsModel;
+use Contao\PageModel;
+use Contao\StringUtil;
+use FeedIo\Feed;
+use FeedIo\FeedInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class FeedProvider implements FeedProviderInterface
+{
+    private ?PageModel $pageModel = null;
+    private ?Request $request = null;
+    private ?string $baseUrl = null;
+
+    public function __construct(private readonly ContaoFramework $framework)
+    {
+    }
+
+    public function setContext(PageModel $pageModel, Request $request, string $baseUrl): void
+    {
+        $this->pageModel = $pageModel;
+        $this->request = $request;
+        $this->baseUrl = $baseUrl;
+    }
+
+    public function getFeed(): FeedInterface
+    {
+        $this->throwIfContextNotSet();
+
+        return (new Feed())
+            ->setTitle($this->pageModel->title)
+            ->setDescription($this->pageModel->description)
+            ->setLanguage($this->pageModel->language)
+        ;
+    }
+
+    public function getItems(): ?Collection
+    {
+        $this->throwIfContextNotSet();
+
+        $archives = StringUtil::deserialize($this->pageModel->newsArchives);
+        $featured = match ($this->pageModel->feedFeatured) {
+            'featured' => true,
+            'unfeatured' => false,
+            default => null,
+        };
+        $newsModel = $this->framework->getAdapter(NewsModel::class);
+
+        if ($this->pageModel->maxFeedItems > 0) {
+            return $newsModel->findPublishedByPids($archives, $featured, $this->pageModel->maxFeedItems, 0, [
+                'return' => 'Collection',
+            ]);
+        }
+
+        return $newsModel->findPublishedByPids($archives, $featured, 0, 0, [
+            'return' => 'Collection',
+        ]);
```

Comment:
Not sure what problem this is solving, but it could probably be simplified to this?

```suggestion
        return $newsModel->findPublishedByPids($archives, $featured, max((int) $this->pageModel->maxFeedItems, 0), 0, [
            'return' => 'Collection',
        ]);
```

Reply:
This code is obsolete since we only use events now.